stdlib/http/src/main/ballerina/http/caching/http_caching_client.bal

```diff
@@ -556,11 +556,14 @@ function getFreshnessLifetime(Response cachedResponse, boolean isSharedCache) re
             if (cachedResponse.hasHeader(DATE)) {
                 string[] dateHeader = cachedResponse.getHeaders(DATE);
 
+                var tExpiresHeader = time:parse(expiresHeader[0], time:TIME_FORMAT_RFC_1123);
+                var tDateHeader = time:parse(dateHeader[0], time:TIME_FORMAT_RFC_1123);
+
                 if (dateHeader.length() == 1) {
-                    // TODO: See if time parsing errors need to be handled
-                    int freshnessLifetime = (time:parse(expiresHeader[0], time:TIME_FORMAT_RFC_1123).time
-                            - time:parse(dateHeader[0], time:TIME_FORMAT_RFC_1123).time) / 1000;
-                    return freshnessLifetime;
+                    if (tExpiresHeader is time:Time && tDateHeader is time:Time) {
+                        int freshnessLifetime = (tExpiresHeader.time - tDateHeader.time) /1000;
+                        return freshnessLifetime;
+                    }
```

Comment:
Shall we add an else clause here as well? The else clause should simply be `return STALE;`

Reply:
IMO we don't need to have else clause here as all these else clauses will be handled by the return statement at the end of the function. 
https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/ballerina/http/caching/http_caching_client.bal#L571

Reply:
Ah yes. Sorry my bad! Totally missed that final return.